src/main/java/spiffe/api/svid/SpiffeWorkloadStub.java

```diff
@@ -0,0 +1,72 @@
+package spiffe.api.svid;
+
+import io.grpc.*;
+import io.grpc.stub.StreamObserver;
+import spiffe.api.svid.SpiffeWorkloadAPIGrpc.SpiffeWorkloadAPIStub;
+
+import static spiffe.api.svid.Workload.*;
+
+/**
+ * Wraps and configures a SpiffeWorkloadAPIBlockingStub with a ManagedChannel
+ * and an ClientInterceptor to add the SECURITY HEADER
+ *
+ */
+class SpiffeWorkloadStub {
+
+    private static final String SECURITY_HEADER = "workload.spiffe.io";
+
+    private SpiffeWorkloadAPIStub workloadAPIAsyncStub;
+
+    /**
+     * Constructor
+     * @param spiffeEndpointAddress where the WorkloadAPI is listening
+     */
+    SpiffeWorkloadStub(String spiffeEndpointAddress) {
+        ManagedChannel managedChannel = SpiffeEndpointChannelBuilder.newChannel(spiffeEndpointAddress);
+
+        workloadAPIAsyncStub= SpiffeWorkloadAPIGrpc
+                                        .newStub(managedChannel)
+                                        .withWaitForReady()
+                                        .withInterceptors(new SecurityHeaderInterceptor());
+    }
+
+    /**
+     * Default constructor
+     * As no 'spiffeEndpointAddress' is provided, the channel builder will resolve it through the Environment
+     */
+    SpiffeWorkloadStub() {
+        this("");
```

Comment:
Just curious, why are you using a blank string instead of null to signal a missing value?

Reply:
yes, you are right, fixed.
thank you